near-sdk/src/environment/mock/external.rs

```diff
@@ -192,11 +197,13 @@ impl External for SdkExternal {
     fn append_action_delete_account(
         &mut self,
         receipt_index: u64,
-        beneficiary_id: String,
+        beneficiary_id: VmAccountId,
     ) -> Result<()> {
-        self.receipts.get_mut(receipt_index as usize).unwrap().actions.push(
-            VmAction::DeleteAccount { beneficiary_id: AccountId::new_unchecked(beneficiary_id) },
-        );
+        self.receipts
+            .get_mut(receipt_index as usize)
+            .unwrap()
```

Comment:
I think we can just change this unwrap to yield an invalid receipt index instead for None
```suggestion
            .ok_or_else(|| HostError::InvalidReceiptIndex { receipt_index })?
```

Reply:
Fair point! I didn't change this in the update and this is just used for testing, but this might yield a better error